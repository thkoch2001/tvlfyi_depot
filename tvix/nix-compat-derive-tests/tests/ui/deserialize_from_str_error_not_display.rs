use std::str::FromStr;

use nix_compat_derive::NixDeserialize;

#[derive(NixDeserialize)]
#[nix(from_str)]
pub struct Test;

impl FromStr for Test {
    type Err = ();
    fn from_str(s: &str) -> Result<Self, Self::Err> {
        if s == "test" {
            Ok(Test)
        } else {
            Err(())
        }
    }
}

fn main() {}
