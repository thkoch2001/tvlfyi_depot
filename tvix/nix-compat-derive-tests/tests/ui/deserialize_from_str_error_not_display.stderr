error[E0277]: `()` doesn't implement `std::fmt::Display`
 --> tests/ui/deserialize_from_str_error_not_display.rs:6:7
  |
6 | #[nix(from_str)]
  |       ^^^^^^^^ `()` cannot be formatted with the default formatter
  |
  = help: the trait `std::fmt::Display` is not implemented for `()`
  = note: in format strings you may be able to use `{:?}` (or {:#?} for pretty-print) instead
note: required by a bound in `invalid_data`
 --> $WORKSPACE/nix-compat/src/wire/de/mod.rs
  |
  |     fn invalid_data<T: fmt::Display>(msg: T) -> Self {
  |                        ^^^^^^^^^^^^ required by this bound in `Error::invalid_data`
